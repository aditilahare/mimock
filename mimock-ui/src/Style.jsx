import React from 'react';
import {
	Button,
	Pill,
	CustomButton,
	IconButtonWithLabel,
	IconButtonWithoutLabel,
	BasicTextField,
	PasswordField,
} from 'styles';
import { PillVariants } from './styles/components/Pill';
import { ButtonVariants, IconButtonVariants } from './styles/components/Button';

function Style() {
	return (
		<div data-testid='style' className='mt-10 text-3xl mx-auto w-screen'>
			<div className='w-full mb-20'>
				<div className='font-sans font-bold text-lg mb-10 ml-6 dark:text-gray-100'>
					Button
				</div>
				<div className='w-full grid grid-cols-4 gap-4 gap-y-8'>
					<Button
						variant={ButtonVariants.RedButton}
						width='w-5/6'
						label='Cancel'
					></Button>
					<Button
						variant={ButtonVariants.GreenButton}
						width='w-1/2'
						label='Success'
					></Button>
					<Button
						variant={ButtonVariants.DarkRedButton}
						label='Cancel'
					></Button>
					<Button
						variant={ButtonVariants.DarkGreenButton}
						width='w-1/2'
						label='Success'
					></Button>
					<Button variant={ButtonVariants.BlueButton} label='Primary'></Button>
					<Button
						variant={ButtonVariants.DefaultButton}
						width='w-3/4'
						label='Default'
					></Button>
					<Button variant={ButtonVariants.TealButton} label='LOGIN'></Button>

					<CustomButton
						label='Custom1'
						color='text-white'
						background='bg-orange-400'
					></CustomButton>
					<CustomButton
						label='Custom2'
						color='text-white'
						background='bg-indigo-300'
					></CustomButton>

					<IconButtonWithLabel
						label='Add'
						width='w-1/3'
						variant={IconButtonVariants.AddButton}
					></IconButtonWithLabel>
					<IconButtonWithLabel
						label='Close'
						width='w-1/2'
						variant={IconButtonVariants.ClearButton}
					></IconButtonWithLabel>
					<IconButtonWithLabel
						label='Delete'
						variant={IconButtonVariants.DeleteButton}
					></IconButtonWithLabel>
					<IconButtonWithLabel
						label='Copy'
						width='w-1/2'
						variant={IconButtonVariants.CopyButton}
					></IconButtonWithLabel>
					<IconButtonWithLabel label='Default'></IconButtonWithLabel>

					<IconButtonWithoutLabel
						variant={IconButtonVariants.AddButton}
					></IconButtonWithoutLabel>
					<IconButtonWithoutLabel
						variant={IconButtonVariants.ClearButton}
					></IconButtonWithoutLabel>
					<IconButtonWithoutLabel
						variant={IconButtonVariants.DeleteButton}
					></IconButtonWithoutLabel>
					<IconButtonWithoutLabel
						variant={IconButtonVariants.CopyButton}
					></IconButtonWithoutLabel>
					<IconButtonWithoutLabel></IconButtonWithoutLabel>
				</div>
			</div>

			<div className='w-full mb-20'>
				<div className='font-sans font-bold text-lg mb-20 ml-6 dark:text-gray-100'>
					Pills
				</div>
				<div className='w-full grid grid-cols-4 gap-4 gap-y-8'>
					<Pill variant={PillVariants.RedPill} label='CANCEL'></Pill>
					<Pill variant={PillVariants.GreenPill} label='SUCCESS'></Pill>
					<Pill variant={PillVariants.YellowPill} label='WARNING'></Pill>
					<Pill variant={PillVariants.DefaultPill} label='DEFAULT'></Pill>
				</div>
			</div>

			<div className='w-full mb-20'>
				<div className='font-sans font-bold text-lg mb-20 ml-6 dark:text-gray-100'>
					Text Field
				</div>
				<div className='w-full grid grid-cols-4 gap-4 gap-y-8 ml-20'>
					<BasicTextField
						label='Username'
						errorLabel='username is required'
						isError={false}
					></BasicTextField>
					<PasswordField></PasswordField>
				</div>
			</div>
		</div>
	);
}

export default Style;
