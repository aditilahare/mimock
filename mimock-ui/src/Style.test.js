import React from 'react';
import { render } from '@testing-library/react';
import Style from './Style.jsx';

describe('Button', () => {
	let tree;

	it('should render style for button and pill', async () => {
		tree = await render(<Style />);

		const { getByTestId, container } = tree;

		expect(getByTestId('style')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});
});
