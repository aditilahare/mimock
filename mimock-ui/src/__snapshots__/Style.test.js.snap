// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render style for button and pill 1`] = `
<div>
  <div
    class="mt-10 text-3xl mx-auto w-screen"
    data-testid="style"
  >
    <div
      class="w-full mb-20"
    >
      <div
        class="font-sans font-bold text-lg mb-20 ml-6"
      >
        Button
      </div>
      <div
        class="w-full flex justify-evenly flex-wrap"
      >
        <button
          class="px-9 py-3 m-auto shadow-md rounded-md transition-all text-center w-fit text-base font-semibold cursor-pointer hover:translate-y-0.5 hover:brightness-110 bg-red-300 text-red-800"
          data-testid="button"
          variantprops="[object Object]"
        >
          Cancel
        </button>
        <button
          class="px-9 py-3 m-auto shadow-md rounded-md transition-all text-center w-fit text-base font-semibold cursor-pointer hover:translate-y-0.5 hover:brightness-110 bg-green-300 text-lime-700"
          data-testid="button"
          variantprops="[object Object]"
        >
          Success
        </button>
        <button
          class="px-9 py-3 m-auto shadow-md rounded-md transition-all text-center w-fit text-base font-semibold cursor-pointer hover:translate-y-0.5 hover:brightness-110 bg-blue-600 text-white"
          data-testid="button"
          variantprops="[object Object]"
        >
          Primary
        </button>
        <button
          class="px-9 py-3 m-auto shadow-md rounded-md transition-all text-center w-fit text-base font-semibold cursor-pointer hover:translate-y-0.5 hover:brightness-110 bg-white text-black"
          data-testid="button"
          variantprops="[object Object]"
        >
          Default
        </button>
      </div>
    </div>
    <div
      class="w-full mb-20"
    >
      <div
        class="font-sans font-bold text-lg mb-20 ml-6"
      >
        Pills
      </div>
      <div
        class="w-full flex flex-wrap justify-evenly"
      >
        <div
          class="px-6 py-3 m-auto shadow-md rounded-full text-sm text-center w-fit border-2 cursor-pointer bg-red-100 text-red-500 border-red-500"
          data-testid="pill"
          variantprops="[object Object]"
        >
          CANCEL
        </div>
        <div
          class="px-6 py-3 m-auto shadow-md rounded-full text-sm text-center w-fit border-2 cursor-pointer bg-green-100 text-green-500 border-green-500"
          data-testid="pill"
          variantprops="[object Object]"
        >
          SUCCESS
        </div>
        <div
          class="px-6 py-3 m-auto shadow-md rounded-full text-sm text-center w-fit border-2 cursor-pointer bg-yellow-100 text-yellow-500 border-yellow-500"
          data-testid="pill"
          variantprops="[object Object]"
        >
          WARNING
        </div>
        <div
          class="px-6 py-3 m-auto shadow-md rounded-full text-sm text-center w-fit border-2 cursor-pointer bg-white text-gray-500 border-gray-500"
          data-testid="pill"
          variantprops="[object Object]"
        >
          DEFAULT
        </div>
      </div>
    </div>
  </div>
</div>
`;
